libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineTiles.kt

```diff
@@ -0,0 +1,104 @@
+package com.mapbox.services.android.navigation.v5.navigation
+
+import com.mapbox.api.routetiles.v1.MapboxRouteTiles
+import com.mapbox.core.exceptions.ServicesException
+import com.mapbox.geojson.BoundingBox
+import okhttp3.ResponseBody
+import retrofit2.Callback
+
+/**
+ * This is a wrapper class for the [MapboxRouteTiles] class. This class takes care of
+ * interfacing with [MapboxRouteTiles] and receives a TAR file wrapped in a ResponseBody
+ * which can be handled using a
+ * [com.mapbox.services.android.navigation.v5.utils.DownloadTask].
+ */
+class OfflineTiles private constructor(
+    private val mapboxRouteTiles: MapboxRouteTiles,
+    private val version: String
+) {
+
+    /**
+     * Call when you have constructed your OfflineTiles object with your desired parameters.
+     * [Callback] must be passed into the method to handle both the response and failure.
+     *
+     * @param callback for retrofit
+     */
+    // FIXME make internal after kotlin migration
+    protected fun fetchRouteTiles(callback: Callback<ResponseBody>) {
+        mapboxRouteTiles.enqueueCall(callback)
+    }
+
+    /**
+     * Returns the version of the current builder.
+     *
+     * @return the version of the current builder
+     */
+    fun version(): String {
```

Comment:
Should we rename this into `getVersion` enabling `.` access notation?

Reply:
so I think it depends on could we break user's API. If we could than I can change this to property access like `(public) val version`

Reply:
As we're doing a completely overhaul for `1.0.0` I think it's reasonable breaking SemVer here. Let's make sure we update the release notes and the android docs site with these changes (and add the `SEMVER` label to the PR 😅) and we're good.